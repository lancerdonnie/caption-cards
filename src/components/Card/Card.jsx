import React from 'react';
import PropTypes from 'prop-types';
import { StyledCard } from '../Styles.jsx';
import Select from '../Select/Select';

const propTypes = {
  caption: PropTypes.string.isRequired
};

const Card = props => {
  return (
    <StyledCard>
      {props.caption}
      <Select captionId={props.id} />
    </StyledCard>
  );
};

Card.propTypes = propTypes;

export default Card;
