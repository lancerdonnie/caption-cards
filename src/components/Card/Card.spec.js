import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import Card from './Card';
import store from '../../redux/store';

afterEach(cleanup);
it('renders', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Card caption={'i pray everyday'} />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
