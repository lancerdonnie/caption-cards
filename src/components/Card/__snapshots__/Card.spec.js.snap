// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ gMrftx"
  >
    i pray everyday
    <div
      class="addselect"
    >
      <span
        class="dropdown"
      >
        <span>
          Add Tag
        </span>
        <span
          class="divider"
        />
        <span>
          <i
            class="fas fa-plus"
          />
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
